cloud/scope/powervs_cluster_test.go

```diff
@@ -1570,3 +1573,415 @@ func getPowerVSClusterScopeParams() PowerVSClusterScopeParams {
+		},
+		{
+			name:           "test DHCP server state error",
+			dhcpServer:     models.DHCPServerDetail{ID: ptr.To("dhcpIDError"), Status: ptr.To(string(infrav1beta2.DHCPServerStateError))},
+			expectedStatus: false,
+		},
+		{
+			name:           "test DHCP server state invalid value",
+			dhcpServer:     models.DHCPServerDetail{ID: ptr.To("dhcpIDError"), Status: ptr.To("InvalidState")},
+			expectedStatus: false,
+		},
+	}
+	for _, tc := range testCases {
+		g := NewWithT(t)
+		clusterScope := PowerVSClusterScope{}
+		t.Run(tc.name, func(_ *testing.T) {
+			status, _ := clusterScope.checkDHCPServerStatus(tc.dhcpServer)
+			g.Expect(status).To(Equal(tc.expectedStatus))
+		})
+	}
+}
+
+func TestCreateDHCPServer(t *testing.T) {
+	var (
+		mockPowerVS *mockP.MockPowerVS
+		mockCtrl    *gomock.Controller
+		clusterName = "clusterName"
+	)
+
+	setup := func(t *testing.T) {
+		t.Helper()
+		mockCtrl = gomock.NewController(t)
+		mockPowerVS = mockP.NewMockPowerVS(mockCtrl)
+	}
+	teardown := func() {
+		mockCtrl.Finish()
+	}
+
+	t.Run("When dhcpServerDetails is empty", func(t *testing.T) {
+		g := NewWithT(t)
+		setup(t)
+		t.Cleanup(teardown)
+
+		dhcpNetwork := &models.DHCPServerNetwork{ID: ptr.To("dhcpNetworkID")}
+		dhcpServer := &models.DHCPServer{ID: ptr.To("dhcpID"), Network: dhcpNetwork}
+		clusterScopeParams := getPowerVSClusterScopeParams()
+		clusterScopeParams.IBMPowerVSCluster.Spec.DHCPServer = nil
+		clusterScopeParams.IBMPowerVSCluster.Name = clusterName
+		clusterScopeParams.PowerVSClientFactory = func() (powervs.PowerVS, error) {
+			mockPowerVS.EXPECT().CreateDHCPServer(gomock.Any()).Return(dhcpServer, nil)
+			return mockPowerVS, nil
+		}
+		clusterScope, _ := NewPowerVSClusterScope(clusterScopeParams)
+		dhcpID, err := clusterScope.createDHCPServer()
+		g.Expect(dhcpID).To(Equal(dhcpServer.ID))
+		g.Expect(err).To(BeNil())
+		g.Expect(clusterScope.IBMPowerVSCluster.Status.Network.ID).To(Equal(dhcpNetwork.ID))
+	})
+
+	t.Run("When dhcpServerDetails are all set properly", func(t *testing.T) {
+		g := NewWithT(t)
+		setup(t)
+		t.Cleanup(teardown)
+
+		dhcpServer := &models.DHCPServer{ID: ptr.To("dhcpID")}
+		clusterScopeParams := getPowerVSClusterScopeParams()
+		clusterScopeParams.IBMPowerVSCluster.Spec.DHCPServer.DNSServer = ptr.To("DNSServer")
+		clusterScopeParams.IBMPowerVSCluster.Spec.DHCPServer.Cidr = ptr.To("10.10.1.10/24")
+		clusterScopeParams.IBMPowerVSCluster.Spec.DHCPServer.Snat = ptr.To(true)
+		clusterScopeParams.IBMPowerVSCluster.Name = clusterName
+		clusterScopeParams.PowerVSClientFactory = func() (powervs.PowerVS, error) {
+			mockPowerVS.EXPECT().CreateDHCPServer(gomock.Any()).Return(dhcpServer, nil)
+			return mockPowerVS, nil
+		}
+		clusterScope, _ := NewPowerVSClusterScope(clusterScopeParams)
+		dhcpID, err := clusterScope.createDHCPServer()
+		g.Expect(dhcpID).To(BeNil())
+		g.Expect(err).ToNot(BeNil())
+	})
+
+	t.Run("When dhcpServerDetails are all set properly", func(t *testing.T) {
```

Comment:
The test case name is repeating. Please set it accordingly.

Reply:
Addressed